### PR TITLE
modem: ubx: fix incoming byte processing

### PR DESCRIPTION
**Pull request by joerchan**
`ret` holds the amount of bytes received from the modem. However during processing of the bytes its value is overwritten by the return value of `modem_ubx_process_received_byte`, in practice discarding all but the first byte read.

To prevent this, store the length in a separate variable.

Handle malloc returning NULL pointer, set err and return from function.

### CODE REVIEW

File: subsys/modem/modem_ubx.c [hunk 1]
```diff
@@ -269,7 +269,8 @@ static void modem_ubx_process_handler(struct k_work *item)
 		return;
 	}
 
-	for (int i = 0; i < ret; i++) {
+	const size_t length = ret;
```
Comment (tomi-font) [inline subsys/modem/modem_ubx.c:272]: Fix for compliance:
```suggestion
	const size_t length = ret;

```

**Review by tomi-font — DISMISSED** [review]

File: subsys/modem/modem_ubx.c (hunk 1 above)
Comment (joerchan) [reply subsys/modem/modem_ubx.c:272]: That was quick :) Thanks.

**Review by tomi-font — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]

**Comment by joerchan** [comment]
Found an unhandled possible malloc null-pointer ref.

File: drivers/gnss/gnss_u_blox_m10.c
Comment (tomi-font) [inline drivers/gnss/gnss_u_blox_m10.c:1]: Nit in the commit message: `this just reduces causes loss of information`

**Review by tomi-font — APPROVED** [review]

File: drivers/gnss/gnss_u_blox_m10.c
Comment (joerchan) [reply drivers/gnss/gnss_u_blox_m10.c:1]: Fixed, thanks.

**Review by bjarki-andreasen — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**